tests/simple/testdata/parse.textproto

```diff
@@ -174,3 +174,478 @@ section {
+  test {
+    name: "const"
+    description: "Check that `const` can be used as a receiver function."
+    expr: "a.const() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "continue"
+    description: "Check that `continue` can be used as a receiver function."
+    expr: "a.continue() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "else"
+    description: "Check that `else` can be used as a receiver function."
+    expr: "a.else() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "for"
+    description: "Check that `for` can be used as a receiver function."
+    expr: "a.for() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "function"
+    description: "Check that `function` can be used as a receiver function."
+    expr: "a.function() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "if"
+    description: "Check that `if` can be used as a receiver function."
+    expr: "a.if() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "import"
+    description: "Check that `import` can be used as a receiver function."
+    expr: "a.import() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "let"
+    description: "Check that `let` can be used as a receiver function."
+    expr: "a.let() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "loop"
+    description: "Check that `loop` can be used as a receiver function."
+    expr: "a.loop() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "package"
+    description: "Check that `package` can be used as a receiver function."
+    expr: "a.package() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "namespace"
+    description: "Check that `namespace` can be used as a receiver function."
+    expr: "a.namespace() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "return"
+    description: "Check that `return` can be used as a receiver function."
+    expr: "a.return() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "var"
+    description: "Check that `var` can be used as a receiver function."
+    expr: "a.var() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "void"
+    description: "Check that `void` can be used as a receiver function."
+    expr: "a.void() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "while"
+    description: "Check that `while` can be used as a receiver function."
+    expr: "a.while() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+}
+section {
+  name: "struct_names"
```

Comment:
We're going to restrict the use of identifiers in type names to declared types, which will render most of the struct_names tests moot. Perhaps remove this even though it's technically correct as it's not feasible to actually execute?